projects/packages/search/src/dashboard/class-initial-state.php

```diff
@@ -71,6 +72,7 @@ public function get_initial_state() {
 				'blogId'            => Jetpack_Options::get_option( 'id', 0 ),
 				'version'           => Package::VERSION,
 				'calypsoSlug'       => ( new Status() )->get_site_suffix(),
+				'postTypeLabels'    => $this->get_post_type_labels(),
```

Comment:
I would recommend a more generic name than `postTypeLabels` maybe just `postTypes`? because in the future we might want to add more information to the objects? say add a local post count, or say other meta data?

Reply:
And the other thing is, the post types is only a mapping from their slugs to the object meta themselves, for which, sorting isn't something we need to care. So in this case, I would prefer an object than an array because which would provide O(1) access straight away, if that makes sense 😄 

```
{
'post' => { 'lable'=>'Posts', ... },
'page' => { 'lable'=>'Posts', ... },
...
}
```

Reply:
> And the other thing is, the post types is only a mapping from their slugs to the object meta themselves, for which, sorting isn't something we need to care. So in this case, I would prefer an object than an array because which would provide O(1) access straight away, if that makes sense 😄


Hmm okay. I understand how O(1) access would be faster, but I'm having a little trouble implementing this in a way that makes sense to me. Do you mean just using an object output in the `get_post_type_labels()` method, but then using it to add back into the array in `record-info.js`? I keep thinking I'm on the right track but then doubting myself with this one in particular 😆

Reply:
Hi @annacmc I added some inline comments, think that'll be clearer 😄

Reply:
thank you @kangzj much clearer, and I totally see why you would recommend this - makes so much easier to access 😄